site/components/pages/Retirements/SingleRetirement/DownloadCertificateButton/generateCertificate.ts

```diff
@@ -0,0 +1,280 @@
+    const formattedRetirementDate = `${retirementDate.getDate()}/${retirementDate.getMonth()}/${retirementDate.getFullYear()}`;
+    let projectDetails = [
+      {
+        label: "Project",
+        value: project?.resourceName,
+      },
+      {
+        label: "Asset Retired",
+        value: params.tokenData.label,
+      },
+      {
+        label: "Retired on",
+        value: formattedRetirementDate,
+      },
+      {
+        label: "Methodology",
+        value: params.retirement.offset.methodology,
+      },
+      {
+        label: "Country/Region",
+        value:
+          params.retirement.offset.country || params.retirement.offset.region,
+      },
+      {
+        label: "Vintage",
+        value: new Date(Number(params.retirement.offset.vintage) * 1000)
+          .getFullYear()
+          .toString(),
+      },
+    ];
+
+    if (isMossRetirement) {
+      projectDetails = [
+        {
+          label: "Asset Retired",
+          value: params.tokenData.label,
+        },
+        {
+          label: "Retired on",
+          value: formattedRetirementDate,
+        },
+      ];
+    }
+
+    const tokenImage = new Image();
+    tokenImage.src = params.tokenData.icon.src;
+    doc.addImage(
+      tokenImage,
+      "JPEG",
+      spacing.tokenImage.x,
+      spacing.tokenImage.y,
+      28,
+      28
+    );
+
+    let startPosition = spacing.projectDetails;
+    projectDetails.forEach((detail) => {
+      const label = `${detail.label}:`;
+      doc.setFont("Poppins", "Bold");
+      doc.text(label, spacing.margin, startPosition);
+
+      doc.setFont("Poppins", "ExtraLight");
+      doc.text(
+        `${detail.value}`,
+        spacing.margin + doc.getTextWidth(label) + 3,
+        startPosition
+      );
+
+      startPosition += 5.5;
+    });
+  };
+
+  const printMossProjectDetails = () => {
+    const linkText = "Learn more ";
+    doc.setFont("Poppins", "Bold");
+    doc.setTextColor(SECONDARY_FONT_COLOR);
+    doc.textWithLink(linkText, spacing.margin, spacing.projectDetails + 11, {
+      url: `${urls.carbonDashboard}/MCO2`,
+    });
+    doc.setTextColor(PRIMARY_FONT_COLOR);
+    doc.text(
+      "about the projects that back the MCO2 pools",
+      spacing.margin + doc.getTextWidth(linkText),
+      spacing.projectDetails + 11
+    );
+  };
+
+  const printRetirementLink = (): void => {
+    const text = "View this retirement on ";
+    doc.setFont("Poppins", "Bold");
+    doc.text(text, spacing.margin, spacing.retirementLink);
+    doc.setTextColor(KLIMA_GREEN);
+    doc.textWithLink(
+      "klimadao.finance",
+      spacing.margin + doc.getTextWidth(text),
+      spacing.retirementLink,
+      { url: params.retirementUrl }
+    );
+  };
+
+  setupFonts();
+  printHeader();
+  printFeatureImage();
+  printRetirementDetails();
+  printTransactionDetails();
+  printProjectDetails();
+  if (isMossRetirement) printMossProjectDetails();
+  printRetirementLink();
+
+  doc.save(`${params.retirement.id}.pdf`);
```

Comment:
Is there a more meaningful file name we can save the certificate name as?

Reply:
retirement_01_0x12345678912381234567.pdf
`retirement_{INDEX}_{BENEFICIARY_ADDRESS}.pdf`